### ISSUE TITLE
Allow automatic customisation and push to trunk

### ISSUE THREAD

**Issue by neonichu** [neonichu/Chocolat#2]
The end goal for this tool is to make the generated podspec ephemeral and just push directly to trunk. For this, any customisations people might want to make to the metadata should be stored in a configuration file and applied automatically as overrides of the defaults. The best way is probably a fragment of a podspec, e.g.

```json
{ "summary": "My override summary", "license": "GPL" }
```

Using an option like `--push` or similar, `chocolat-cli` would invoke `pod trunk push` and the project would be available via CP with minimal user intervention.

**Comment by orta** [comment]
FWIW, there is a GitHub API for getting the license of a project, but then it becomes GitHub specific :-/

**Comment by neonichu** [comment]
There's already some GitHub specific behavior, which I think is a fine default, the override method can provide the 20% solution.

**Comment by alloy** [comment]
Same applies to project description, no?

**Comment by neonichu** [comment]
Yep, I was already planning to the description from :octocat:

**Comment by orta** [comment]
There's a TOML parser in SPM, so it's also possible that their machine readable metadata section could include these bits too

**Comment by neonichu** [comment]
Yah, I think inclusion of "summary" and "license" attributes into `Package.swift` would be something to lobby for :)

**Comment by ddunbar** [comment]
Both of those attributes are things we definitely agree should be in the manifest, just haven't been added yet. I'd be happy to see patches for both -- we will need to go through some discussion / proposal w.r.t. the exact naming, documentation, definition of schema (is license a url? a name? known identifiers?) etc. but I think it is something we could get soon...

**Comment by alloy** [comment]
@ddunbar In our case it’s a bit of a freeform field: https://guides.cocoapods.org/syntax/podspec.html#license.

Where possible we try to steer the spec author to be as specific as possible about the license, but there are so many licenses and also adjustments that people make, that was too hard to completely lock down to a specific list of license names.

While we do try to automatically load license text from a license file, one thing we specifically do **not** do is try to inflect a license name from license content. People can and will make adjustments to licenses, so it’s legally too dangerous for us to try to be too smart about this.

**Comment by neonichu** [comment]
I recently saw that NPM uses a semantic way for expressing licenses, see <https://docs.npmjs.com/files/package.json#license> — that still offers a way to specify any custom license, while making common ones accessible in a programming fashion.

**Comment by alloy** [comment]
@neonichu That’s pretty neat.

**Comment by ddunbar** [comment]
Yeah, that is along the lines of what I would tend to propose. Make it easy for people to do the "right" thing (use a standard license). Python "kinda" did this, as well:
  https://pypi.python.org/pypi?:action=list_classifiers
(see known "License ::" specifiers).
